### PR TITLE
feat(rest) asynchronous end point for downloading reports.

### PR DESCRIPTION
**Pull request by keerthi-bl**
[//]: # (Copyright Bosch.IO GmbH 2020)
[//]: # (This program and the accompanying materials are made)
[//]: # (available under the terms of the Eclipse Public License 2.0)
[//]: # (which is available at https://www.eclipse.org/legal/epl-2.0/)
[//]: # (SPDX-License-Identifier: EPL-2.0)

> Please provide a summary of your changes here.
> * Which issue is this pull request belonging to and how is it solving it? (*Refer to issue here*)
#2023 

> * Did you add or update any new dependencies that are required for your change?
No

Issue: 

Use postman for testing.

Sample URL : 
 http://localhost:8080/resource/api/reports?withlinkedreleases=true&mimetype=xlsx&mailrequest=true&module=components

Sample response:

{
"response": "E-mail sent succesfully to the end user.",
"url": "http://127.0.0.1:8080/resource/api/reports/download?user=keerthi.bl@sw360.org&module=components&extendedByReleases=true&token=/tmp/keerthi.bl@sw360.org/2023-05-17_a990c489-acf8-4600-911c-2d89015b8bba"
}

> Have you implemented any additional tests?

No

### Checklist
Must:
- [ ] All related issues are referenced in commit messages and in PR

### CODE REVIEW

**Comment by keerthi-bl** [comment]
It can be merged only after #1965 . Since its a depended API.

**Comment by keerthi-bl** [comment]
Rebase completed

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java [hunk 1]
```diff
@@ -184,27 +157,28 @@ private boolean validateMimeType(String mimeType) {
...
+            case PROJECTS:
+                buffer = sw360ReportService.getReportStreamFromURl(user, Boolean.valueOf(extendedByReleases), token);
+                break;
+            case COMPONENTS:
+                fileConstant = "components-%s.xlsx";
+                buffer = sw360ReportService.getComponentReportStreamFromURl(user, Boolean.valueOf(extendedByReleases),
+                        token);
+                break;
+            default:
+                break;
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java:180]: buffer might be null

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java [hunk 2]
```diff
@@ -184,27 +157,28 @@ private boolean validateMimeType(String mimeType) {
...
+    @GetMapping(value = REPORTS_URL + "/download")
+    public void downloadExcel(HttpServletRequest request, HttpServletResponse response) throws TException {
+        final User sw360User = restControllerHelper.getSw360UserFromAuthentication();
+        String module = request.getParameter("module");
         String token = request.getParameter("token");
         String extendedByReleases = request.getParameter("extendedByReleases");
-        User user=restControllerHelper.getUserByEmail(sw360User.getEmail());
-        String fileConstant="projects-%s.xlsx";
+        User user = restControllerHelper.getUserByEmail(sw360User.getEmail());
+        String fileConstant = "projects-%s.xlsx";
```
Comment (hoangnt2) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java:167]: Can be replaced by
`String fileConstant = module + "-%s.xlsx"`;

After that line 175 `fileConstant = "components-%s.xlsx";` can be removed

**Comment by keerthi-bl** [comment]
Resolved review comments.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java (hunk 2 above)
Comment (hoangnt2) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java:167]: @keerthi-bl, sorry but I mean replace line 167 by `String fileConstant = module + "-%s.xlsx";` and remove line 175 `fileConstant = "components-%s.xlsx";`

Comment (keerthi-bl) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/report/SW360ReportController.java]: Sorry, yes I updated the same.

**Comment by hoangnt2** [comment]
Look good

**Comment by hoangnt2** [comment]
Look good

**Comment by hoangnt2** [comment]
Look good

**Comment by hoangnt2** [comment]
Look good

**Comment by rudra-superrr** [comment]
Testing was successful.

**Pull request merged**